### PR TITLE
ci: add rook-ceph cluster ci test

### PR DESCRIPTION
**Pull request by subhamkrai**
# Describe what this PR does #

Provide some context for the reviewer

## Is there anything that requires special attention ##

Do you have any questions?

Is the change backward compatible?

Are there concerns around backward compatibility?

Provide any external context for the change, if any.

For example:

* Kubernetes links that explain why the change is required
* Ceph-CSI spec related changes/catch-up that necessitates this patch
* golang related practices that necessitates this change

## Related issues ##

Mention any github issues relevant to this PR. Adding below line
will help to auto close the issue once the PR is merged.

Fixes: #issue_number

## Future concerns ##

List items that are not part of the PR and do not impact it's
functionality, but are work items that can be taken up subsequently.

**Checklist:**

* [ ] **Commit Message Formatting**: Commit titles and messages follow
  guidelines in the [developer
  guide](https://github.com/ceph/ceph-csi-operator/blob/devel/docs/development-guide.md#commit-messages).
* [ ] Reviewed the developer guide on [Submitting a Pull
  Request](https://github.com/ceph/ceph-csi-operator/blob/devel/docs/development-guide.md#development-workflow)
* [ ] [Pending release
  notes](https://github.com/ceph/ceph-csi-operator/blob/devel/PendingReleaseNotes.md)
  updated with breaking and/or notable changes for the next major release.
* [ ] Documentation has been updated, if necessary.
* [ ] Unit tests have been added, if necessary.
* [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by subhamkrai** [comment]
Testing looks good @Madhu-1. Let me what else I need in this PR
```
runner@fv-az1426-553:~/work/ceph-csi-operator/ceph-csi-operator$ kubectl get pods -n rook-ceph

NAME                                                READY   STATUS        RESTARTS   AGE
csi-cephfsplugin-provisioner-76b7548bfb-jw6qr       5/5     Running       0          63s
csi-cephfsplugin-wltc4                              2/2     Running       0          63s
csi-rbdplugin-5zppz                                 2/2     Running       0          63s
csi-rbdplugin-provisioner-744cd895d5-tbd5n          5/5     Running       0          63s
rook-ceph-exporter-fv-az1426-553-5c5d75bb48-77cvb   1/1     Terminating   0          33s
rook-ceph-mgr-a-77f5d6dcc7-cf5z8                    1/1     Running       0          33s
rook-ceph-mon-a-c7cbff6f7-t5g9n                     1/1     Running       0          82s
rook-ceph-mon-b-5455d8fb77-46px5                    1/1     Running       0          58s
rook-ceph-mon-c-675b896c-hcfd2                      1/1     Running       0          47s
rook-ceph-operator-757cdc49bb-p2j47                 1/1     Running       0          2m8s
rook-ceph-osd-0-5765bf69d8-ckvnq                    0/1     Init:0/3      0          3s
rook-ceph-osd-prepare-fv-az1426-553-6d9sq           0/1     Completed     0          11s
runner@fv-az1426-553:~/work/ceph-csi-operator/ceph-csi-operator$ 
runner@fv-az1426-553:~/work/ceph-csi-operator/ceph-csi-operator$ kubectl  get cephcluster -n rook-ceph
NAME         DATADIRHOSTPATH   MONCOUNT   AGE    PHASE   MESSAGE                        HEALTH      EXTERNAL   FSID
my-cluster   /var/lib/rook     3          2m1s   Ready   Cluster created successfully   HEALTH_OK              eb96c8fc-b615-4c64-a3f8-ea1143dd528c
runner@fv-az1426-553:~/work/ceph-csi-operator/ceph-csi-operator$ 
```

```
runner@fv-az1426-553:~/work/ceph-csi-operator/ceph-csi-operator$ kubectl  -n rook-ceph exec -it rook-ceph-tools-58c6857df4-vbj55 -- sh
sh-4.4$ ceph -s
  cluster:
    id:     eb96c8fc-b615-4c64-a3f8-ea1143dd528c
    health: HEALTH_OK
 
  services:
    mon: 3 daemons, quorum a,b,c (age 5m)
    mgr: a(active, since 3m)
    osd: 1 osds: 1 up (since 4m), 1 in (since 4m)
 
  data:
    pools:   1 pools, 32 pgs
    objects: 2 objects, 463 KiB
    usage:   27 MiB used, 75 GiB / 75 GiB avail
    pgs:     32 active+clean
```

File: .github/workflows/set_rook_ceph_cluster.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: rook-ceph-cluster
```
Comment (Madhu-1) [inline .github/workflows/set_rook_ceph_cluster.yaml:1]: make this as just ceph cluster? and also update the file name to be `setup_ceph_cluster?`

File: .github/workflows/tmate_debug/action.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: "Tmate debugging tests"
```
Comment (Madhu-1) [inline .github/workflows/tmate_debug/action.yml:1]: which this action is used?

File: .github/workflows/set_rook_ceph_cluster.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      - name: print k8s cluster status
+        run: |
+            minikube status
+            kubectl get nodes
+
+      - name: use local disk
+        run: test/scripts/github-action-helper.sh use_local_disk
+
+      - name: deploy rook cluster
+        run: test/scripts/github-action-helper.sh deploy_rook
```
Comment (Madhu-1) [inline .github/workflows/set_rook_ceph_cluster.yaml:52]: name this as `create ceph cluster`

File: test/scripts/github-action-helper.sh [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+  sudo swapoff --all --verbose
+  sudo umount /mnt
+  # search for the device since it keeps changing between sda and sdb
+  sudo wipefs --all --force "$BLOCK_DATA_PART"
+}
+
+deploy_rook() {
+  kubectl create -f https://raw.githubusercontent.com/rook/rook/master/deploy/examples/common.yaml
+  kubectl create -f https://raw.githubusercontent.com/rook/rook/master/deploy/examples/crds.yaml
+  kubectl create -f https://raw.githubusercontent.com/rook/rook/master/deploy/examples/operator.yaml
```
Comment (Madhu-1) [inline test/scripts/github-action-helper.sh:42]: use released version of Rook

File: test/scripts/github-action-helper.sh [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+  wait_for_operator_pod_to_be_ready_state
+  curl https://raw.githubusercontent.com/rook/rook/master/deploy/examples/cluster-test.yaml -o cluster-test.yaml
```
Comment (Madhu-1) [inline test/scripts/github-action-helper.sh:44]: disable the cephcsi driver deployed by Rook.

File: .github/workflows/set_rook_ceph_cluster.yaml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        if: ${{ failure() || runner.debug || contains(github.event.pull_request.labels.*.name, 'debug-ci') }}
+        uses: mxschmitt/action-tmate@v3
+        with:
+          limit-access-to-actor: false
+          detached: true
+
+
+      - name: Setup Minikube
+        run: |
+            test/scripts/github-action-helper.sh install_minikube_with_none_driver v1.28.4
```
Comment (Madhu-1) [inline .github/workflows/set_rook_ceph_cluster.yaml:41]: set up the latest released version of kubernetes and also make the minikube_version as configurable here so we don't need to look into multiple places to change the configurations

**Review by Madhu-1 — CHANGES_REQUESTED** [review]

File: .github/workflows/set_rook_ceph_cluster.yaml (hunk 1 above)
Comment (subhamkrai) [reply .github/workflows/set_rook_ceph_cluster.yaml]: done

File: .github/workflows/tmate_debug/action.yml (hunk 2 above)
Comment (subhamkrai) [reply .github/workflows/tmate_debug/action.yml]: removed

File: .github/workflows/set_rook_ceph_cluster.yaml (hunk 3 above)
Comment (subhamkrai) [reply .github/workflows/set_rook_ceph_cluster.yaml]: done

File: test/scripts/github-action-helper.sh (hunk 4 above)
Comment (subhamkrai) [reply test/scripts/github-action-helper.sh]: done

File: test/scripts/github-action-helper.sh (hunk 5 above)
Comment (subhamkrai) [reply test/scripts/github-action-helper.sh]: done

File: .github/workflows/set_rook_ceph_cluster.yaml (hunk 6 above)
Comment (subhamkrai) [reply .github/workflows/set_rook_ceph_cluster.yaml]: done

File: .github/workflows/set_rook_ceph_cluster.yaml [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    # reference: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell
+    shell: bash --noprofile --norc -eo pipefail -x {0}
+
+# cancel the in-progress workflow when PR is refreshed.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  rook-ceph-cluster:
```
Comment (Madhu-1) [inline .github/workflows/set_rook_ceph_cluster.yaml:19]: can we change this to ceph-cluster?

Comment (Madhu-1) [reply .github/workflows/set_rook_ceph_cluster.yaml:19]: can you also update the file name to have `setup_ceph_cluster`?

**Review by Madhu-1 — COMMENTED** [review]
small nits, LGTM

File: .github/workflows/set_rook_ceph_cluster.yaml (hunk 7 above)
Comment (subhamkrai) [reply .github/workflows/set_rook_ceph_cluster.yaml]: done

**Review by Madhu-1 — APPROVED** [review]

**Pull request merged**